doc/source/conf.py

```diff
@@ -70,9 +70,12 @@
 
 # The theme to use for HTML and HTML Help pages.  See the documentation for
 # a list of builtin themes.
-#
+
 html_theme = "furo"
 
+# custom title
+html_title = "Software version " + release
```

Comment:
You can simplify even more:
```suggestion
html_title = "Version " + release
```
"Software" sounds a bit verbose and generic ("package" would be more specific, but just "version" is already enough).